pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -2162,11 +2162,6 @@ private PersistentReplicator getReplicatorReference(String replName, PersistentT
         String path = path(PARTITIONED_TOPIC_PATH_ZNODE, topicName.getPersistenceNamingEncoding());
         CompletableFuture<Void> result = new CompletableFuture<>();
         pulsar().getBrokerService().fetchPartitionedTopicMetadataAsync(topicName).thenAccept(partitionMetadata -> {
-            if (partitionMetadata.partitions <= 1) {
```

Comment:
Shouldn't this be changed into `< 1` ?

Reply:
partitionMetadata is the current metadata for the topic.  If we check "< 1" that would be saying for any topic that has greater than one partition, it cannot be updated and the exception will be returned.

Reply:
No, it means to throw exception if `partitionMetadata.partitions == 0` (which is the case when topic is not partitioned), though not throwing it when `partitionMetadata.partitions == 1` (as we're incorrectly doing now) or when `partitionMetadata.partitions > 1` (which is already correctly handled).

Reply:
will fix